src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -287,6 +265,29 @@ void GinIndexStore::cancel() noexcept
         postings_file_stream->cancel();
 }
 
+void GinIndexStore::initSegmentId()
+{
+    String segment_id_file_name = getName() + GIN_SEGMENT_ID_FILE_TYPE;
+
+    UInt32 result;
+    if (storage->existsFile(segment_id_file_name))
+    {
+        std::unique_ptr<DB::ReadBufferFromFileBase> istr = this->storage->readFile(segment_id_file_name, {}, std::nullopt, std::nullopt);
+
+        uint8_t version = 0;
+        readBinary(version, *istr);
+
+        if (version > static_cast<std::underlying_type_t<Format>>(CURRENT_GIN_FILE_FORMAT_VERSION))
```

Comment:
As of now, no mechanism to support older file formats is implemented. I think we check for equivalence here, i.e. only support the current format version. (also in l. 212)